### ISSUE TITLE
DVQA: Understanding Data Visualizations via Question Answering

### ISSUE THREAD

**Issue by howardyclo** [howardyclo/papernotes#32]
### Metadata
- Authors: Kushal Kafle, Scott Cohen, +1 author Christopher Kanan
- Organization: Rochester Institute of Technology & Adobe Research
- Conference: CVPR 2018
- Paper: https://arxiv.org/pdf/1801.08163.pdf
- Code: https://github.com/kushalkafle/DVQA_dataset

**Comment by howardyclo** [comment]
### Summary
- This paper presents "DVQA", a challenging VQA dataset for bar chart (structure, data retrieval and reasoning).
- This paper demonstrates exsiting VQA algorithms can not perform well on DVQA.
- This paper purposes 2 strong baselines that perform considerably better.

### Differences from Existing VQA
- Existing VQA mostly operates on natural images.
- Existing VQA assumes 2 fixed vocabulary for encode Q and A, thus cannot deal with out-of-vocabulary (OOV) words. (This could be solved by pointer network like #30)
- Small changes in an natural image typically have little impact on a question in VQA, while in chart images, even small changes can completely alter the information in the chart.
- DVQA requires "reading" text from a chart to correctly answer them. 

### Differences from Concurrent Work — FigureQA
- It contains only yes/no type questions.
- It does not contain questions that require numeric values as answers.
- It has fixed labels for bars across different figures.  (e.g., a red bar is always labeled ’red’).
- It avoids the OOV problem.

### Dataset Statistics 
3,487,194 total question answer pairs for 300,000 images divided into three major question types.

![](https://github.com/howardyclo/papernotes/blob/master/images/DVQA%20Understanding%20Data%20Visualizations%20via%20Question%20Answering/1.png?raw=true)

![](https://github.com/howardyclo/papernotes/blob/master/images/DVQA%20Understanding%20Data%20Visualizations%20via%20Question%20Answering/2.png?raw=true)

### Dataset Synthesis Process

![](https://github.com/howardyclo/papernotes/blob/master/images/DVQA%20Understanding%20Data%20Visualizations%20via%20Question%20Answering/3.png?raw=true)

- Select a visual style for a chart, choosing data for a chart, and then generating questions for that chart.
- Plotting tool: Matplotlib.
- Visual factors:
    - Number of bars and groups.
    - Presence or absence of grid lines.
    - Difference in color, width, spacing, orientation, and texture of the bars.
    - Difference in the orientation and the location of labels and legends.
- Select the 1000 most frequent nouns in the Brown Corpus using NLTK’s part-of-speech tagging for our training set and our ‘easy’ test set **Test-Familiar**. To measure a system’s ability to scale to unknown answers, we also created a more difficult test set **Test-Novel**, in which we use 500 new words that are not seen during training.
- Data scales: Linear (1-10), percentage (10-100), and exponential (1-10^10).  A small percentage of bars are allowed to have a value of zero which appears as a missing bar in the chart.
- Generate questions from fixed template.
- Balance answers to minimize dataset bias.

### Baselines
- YES: This model answers 'YES' for all questions. 
- IMG: A question-blind model. ResNet-152 with 448 × 448 images + 1-layer 1024 MLP.
- QUES: A image-blind model. 1-layer 1024 LSTM + 1-layer 1024 MLP.
- IMG+QUES: Concat CNN+LSTM features + 1-layer 1024 MLP.
- SAN-VQA: The [Stacked Attention Network (SAN)](https://www.youtube.com/watch?v=nlcvZZWhoL8) for VQA. (State-of-the-art on VQA 1.0 and 2.0).

### Purposed Model 1: Multi-Output Model (MOM)

![](https://github.com/howardyclo/papernotes/blob/master/images/DVQA%20Understanding%20Data%20Visualizations%20via%20Question%20Answering/4.png?raw=true)

- Two sub-network perform classification or OCR to answer different kinds of questions.
- One separate binary classifier that determines which of the outputs to trust.
- Problems:  It has no explicit ability to visually read bar chart text and its LSTM question encoding cannot handle chart-specific words. (It just encode image regions from OCR instead of strings)

### Purposed Model 2:  SAN with DYnamic Encoding Model (SANDY)
- OCR gives us the bounding boxes and corresponding strings.
- Assign unique IDs to the bounding boxes.
- 2 versions:
    - One assumes OCR's output is perfect (from dataset annotation)
    - The other one uses the output of open-source Tesseract OCR. Tesseract’s output is pre-processed in three ways: 1) we only use words with alphabetical characters in them, 2) we filter word detections with confidence less than 50%, and 3) we filter single-character word detections.

#### Training
- All of the classification based systems, except SANDY and the OCR branch of MOM, use a global answer dictionary from training set containing 1076 words, so they each have 1076 output units.

#### To Be Continue

**Comment by mineshmathew** [comment]
hey, 
Do you know why the L word global dictionary for SANDY  has just 77 words ( 107 -30) ? MOMs classification branch and other baseline models use a much larger set of classes (1000 +). And how are these words/classes selected ?
